kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/evm-int-simplification.k

```diff
@@ -52,4 +53,20 @@ module EVM-INT-SIMPLIFICATION-COMMON
       requires 0 <Int Y
       [simplification, comm, preserves-definedness]
 
+  // ###########################################################################
+  // chop
+  // ###########################################################################
+
+  rule [chop-idempotent]:
+    chop ( chop ( X:Int ) ) => chop ( X )
+    [simplification]
```

Comment:
I'm surprised this rule is needed. I guess I would think that these three rules together would imply the same: https://github.com/runtimeverification/evm-semantics/blob/6f86c28def0ac17b0998352ec06875229d6bfee8/kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/lemmas.k#L25.

Perhaps we need to mark the third rule there as `smt-lemma` then we won't need this one as well?

Reply:
Maybe also those lemmas linked above could be moved over here.

Reply:
I will move the lemmas and add `smt-lemma`, I think you are correct.